kingpin/actors/rightscale/server_array.py

```diff
@@ -712,14 +719,40 @@ def _execute(self):
                 task=task,
                 task_name=task_name,
                 sleep=self.option('expected_runtime'),
-                logger=self.log.info
+                logger=self.log.info,
+                meta_data=instance
             ))
 
         self.log.info('Waiting for %s tasks to finish.' % count)
-        success = yield task_waiting
+        status_instances = yield task_waiting
```

Comment:
I'm not sure how this can be refactored... You're thinking just 1 method that does "check success of every instance and print logs if not successful" ? There are honestly only 2 lines that do work - 743 and 750 -- the rest is just fluff for logs